server/modules/record/service/update/recordUsersMap.js

```diff
@@ -3,45 +3,57 @@ const R = require('ramda')
 const User = require('../../../../../common/user/user')
 const { isDateBefore } = require('../../../../../common/dateUtils')
 
-const recordUserUuidsMap = new Map()
-const previewRecordsMap = new Map()
+const userUuidsByRecordUuid = new Map()
+const previewRecordsByRecordUuid = new Map()
 
 const getUserUuids = recordUuid =>
   R.defaultTo(
     [],
-    recordUserUuidsMap.get(recordUuid)
+    userUuidsByRecordUuid.get(recordUuid)
   )
 
+const getRecordUuid = userUuid => {
+  for (const recordUuid of userUuidsByRecordUuid.keys()) {
+    const userUuids = userUuidsByRecordUuid.get(recordUuid)
+    if (userUuids && userUuids.has(userUuid)) {
+      return recordUuid
+    }
+  }
+  return null
+}
+
```

Comment:
Can you use find here? i.e.

`
const getRecordUuid = userUuid =>
  userUuidsByRecordUuid.keys().find(recordUuid => {
    const userUuids = userUuidsByRecordUuid.get(recordUuid)
    return userUuids && userUuids.has(userUuid)
  })
`

Reply:
right, of course!